File: app/src/main/java/com/akexorcist/lovelyrecyclerview/adapter/holder/ButtonViewHolder.java
package com.akexorcist.lovelyrecyclerview.adapter.holder;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.Button;

import com.akexorcist.lovelyrecyclerview.R;


/**
 * Created by Akexorcist on 10/28/2016 AD.
 */

public class ButtonViewHolder extends RecyclerView.ViewHolder {
    public Button btnPositive;
    public Button btnNegative;

    public ButtonViewHolder(View itemView) {
        super(itemView);
        btnPositive = (Button) itemView.findViewById(R.id.btn_positive);
        btnNegative = (Button) itemView.findViewById(R.id.btn_negative);
    }
}

File: app/src/main/java/com/akexorcist/lovelyrecyclerview/adapter/holder/ButtonViewHolder.kt
package com.akexorcist.lovelyrecyclerview.adapter.holder

import androidx.recyclerview.widget.RecyclerView
import com.akexorcist.lovelyrecyclerview.databinding.ViewButtonBinding

class ButtonViewHolder(private val binding: ViewButtonBinding) : RecyclerView.ViewHolder(binding.root) {
    fun bind(
        onPositiveClicked: () -> Unit,
        onNegativeClicked: () -> Unit
    ) {
        binding.btnPositive.setOnClickListener { onPositiveClicked.invoke() }
        binding.btnNegative.setOnClickListener { onNegativeClicked.invoke() }
    }
}
